app/components/public/call-for-speakers.js

```diff
@@ -13,7 +13,13 @@ export default class CallForSpeakers extends Component {
     if (this.session.isAuthenticated) {
       this.router.transitionTo('public.cfs.new-speaker');
     } else {
-      this.set('isLoginModalOpen', true);
+      this.flashMessages.add({
+        message           : 'In order to add Speaker details you need to login',
+        type              : 'info',
+        preventDuplicates : true
+      });
+      this.router.transitionTo('login');
+      // this.set('isLoginModalOpen', true);
```

Comment:
Why?

Reply:
when you click on add speaker details button it shows flash message on the same page itself because of the appearance of that pop window(which is triggered by this.set('isLoginModalOpen', true)) . So I have to redirect it to Login page. before the flash message is shown.

Reply:
Do you think we added a modal just for fun? It was there for a reason, we don't want to redirect to the login page from here. Revert

Reply:
So if there is already a modal which ask to login than should I show message on that modal or is there really a need to show any alert message there ?

Reply:
Yes, please show this in modal. Easiest would be to show the message in login component so it works on both the login page and modal

Reply:
1 - 
![addspeaker](https://user-images.githubusercontent.com/57146965/99964250-0457e180-2db9-11eb-9323-fb5b62b176e0.gif)
2 -
![AddSpeakerOnLogin](https://user-images.githubusercontent.com/57146965/99964256-07eb6880-2db9-11eb-8dda-a5f300b46f4e.gif)
which one should I use ?

Reply:
Whichever requires the least amount of coding changes